app/lib/oidc_client.rb

```diff
@@ -0,0 +1,131 @@
+require "openid_connect"
+
+class OidcClient
+  class OAuthFailure < RuntimeError; end
+
+  attr_reader :client_id,
+              :provider_uri
+
+  delegate :authorization_endpoint,
+           :token_endpoint,
+           :userinfo_endpoint,
+           :end_session_endpoint,
+           to: :discover
+
+  def initialize(provider_uri, client_id, secret)
+    @provider_uri = provider_uri
+    @client_id = client_id
+    @secret = secret
+  end
+
+  def auth_uri
+    nonce = SecureRandom.hex(16)
+    {
+      uri: client.authorization_uri(
+        scope: scopes,
+        state: nonce,
+        nonce: nonce,
+      ),
+      state: nonce,
+    }
+  end
+
+  def redirect_uri
+    host =
+      if ENV["VCAP_APPLICATION"].present?
+        "https://#{JSON.parse(ENV['VCAP_APPLICATION'])['uris'].first}"
+      else
+        "http://#{ENV['VIRTUAL_HOST']}"
+      end
+    host + Rails.application.routes.url_helpers.transition_checker_new_session_callback_path
+  end
+
+  def scopes
+    %i[email transition_checker]
+  end
+
+  def callback(code, state)
+    client.authorization_code = code
+    access_token = client.access_token!
+    id_token = OpenIDConnect::ResponseObject::IdToken.decode access_token.id_token, discover.jwks
+    id_token.verify! client_id: client_id, issuer: discover.issuer, nonce: state
+    {
+      access_token: access_token,
+      user_info: access_token.userinfo!,
+    }
+  end
+
+  def get_checker_attribute(access_token:, refresh_token:)
+    response = oauth_request(
+      access_token: access_token,
+      refresh_token: refresh_token,
+      method: :get,
+    )
+
+    if response[:result].empty?
+      response.merge(result: [])
+    else
+      response.merge(result: JSON.parse(response[:result])["claim_value"])
+    end
+  end
+
+  def update_checker_attribute(access_token:, refresh_token:, data:)
+    oauth_request(
+      access_token: access_token,
+      refresh_token: refresh_token,
+      method: :put,
+      arg: { "value" => data.to_json },
+    ).except(:result)
+  end
```

Comment:
This can be removed, so the `method` and `arg` parameters of `oauth_request` can be inlined